codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/serialize/XmlBindingTraitSerializerGenerator.kt

```diff
@@ -191,11 +192,67 @@ class XmlBindingTraitSerializerGenerator(
     }
 
     override fun serverOutputSerializer(operationShape: OperationShape): RuntimeType {
-        TODO("Not yet implemented")
+        val fnName = symbolProvider.serializeFunctionName(operationShape)
+        val outputShape = operationShape.outputShape(model)
+        val xmlMembers = operationShape.operationXmlMembers()
+        val operationXmlName = xmlIndex.operationOutputShapeName(operationShape)
+            ?: throw CodegenException("operation must have a name if it has members")
+        return RuntimeType.forInlineFun(fnName, operationSerModule) {
+            it.rustBlockTemplate(
+                "pub fn $fnName(output: &#{target}) -> Result<String, #{Error}>",
+                *codegenScope, "target" to symbolProvider.toSymbol(outputShape)
+            ) {
+                rust("let mut out = String::new();")
+                // create a scope for writer. This ensure that writer has been dropped before returning the
+                // string and ensures that all closing tags get written
+                rustBlock("") {
+                    rustTemplate(
+                        """
+                        let mut writer = #{XmlWriter}::new(&mut out);
+                        ##[allow(unused_mut)]
+                        let mut root = writer.start_el(${operationXmlName.dq()})${outputShape.xmlNamespace(root = true).apply()};
+                        """,
+                        *codegenScope
+                    )
+                    serializeStructure(outputShape, xmlMembers, Ctx.Element("root", "output"))
+                }
+                rustTemplate("Ok(out)", *codegenScope)
+            }
+        }
     }
 
     override fun serverErrorSerializer(shape: ShapeId): RuntimeType {
-        TODO("Not yet implemented")
+        val errorShape = model.expectShape(shape, StructureShape::class.java)
+        val xmlMembers = httpBindingResolver.errorResponseBindings(shape)
+            .filter { it.location == HttpLocation.DOCUMENT }
+            .map { it.member }
+        val fnName = symbolProvider.serializeFunctionName(errorShape)
+        return RuntimeType.forInlineFun(fnName, operationSerModule) {
+            it.rustBlockTemplate(
+                "pub fn $fnName(error: &#{target}) -> Result<String, #{Error}>",
+                *codegenScope, "target" to symbolProvider.toSymbol(errorShape)
+            ) {
+                rust("let mut out = String::new();")
+                // create a scope for writer. This ensure that writer has been dropped before returning the
+                // string and ensures that all closing tags get written
+                rustBlock("") {
+                    rustTemplate(
+                        """
+                        let mut writer = #{XmlWriter}::new(&mut out);
+                        ##[allow(unused_mut)]
+                        let mut root = writer.start_el("Error")${errorShape.xmlNamespace(root = true).apply()};
```

Comment:
Is there a better way I should use to write the `"Error"`tag or it makes sense to hard-quote it here in the code template?

Reply:
Probably someone working on the client codegen can answer to this..

Reply:
Should we tag someone to ask for review?

Reply:
@rcoh do you mind to help us here?

Reply:
hardcoding `Error` appears to be correct here: https://awslabs.github.io/smithy/1.0/spec/aws/aws-restxml-protocol.html#error-response-serialization.

Note that you'll also need to handle the conditional error wrapping here. On the client side, we handle it by injecting a different error extractor but on the server side, I think you'll need to handle it slightly differently. The most parallel way would be to have a function that created an `ErrorWriter` that could be swapped as a runtime type, but I'll leave that to @crisidev to decide.

https://github.com/awslabs/smithy-rs/blob/1fa0ecfccef7e73a2886937cc1bb7935c4bf4ec0/codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/RestXml.kt#L63-L66

Reply:
In the server there is no need for the injection. We handle an error coming from here as a [serialization rejections](https://github.com/awslabs/smithy-rs/blob/main/rust-runtime/aws-smithy-http-server/src/rejection.rs#L29-L34).